### PR TITLE
Add postgresql13 documentation

### LINKED ISSUE #711: Update to postgresql13
We need to update this page by adding a 12 to 13 migration:

https://www.uyuni-project.org/uyuni-docs/uyuni/upgrade/server-major-upgrade-uyuni.html

and then add a new page for 12 to 13 migration, similar to https://www.uyuni-project.org/uyuni-docs/uyuni/upgrade/db-migration-12.html

Note, the 10 to 12 migration content is still valid.

Posgresql13 is for the upcoming SLE15SP3 while postgresql12 is for the current leap15.2, so we need to keep both migration instructions.

### PR DESCRIPTION
**Pull request by jordimassaguerpla**
# Description

Add postgresql13 documentation

Note postgresql13 works on SLE15SP3 (SUSE Manager 4.2) but for Leap 15.2
we still use postgresql12 (current uyuni version).

# Target branches

Which documentation version does this PR apply to?

- [X] Master (Default)
- [ ] Manager-4.1
- [ ] Manager-4.0

# Links

Fixes #711

### CODE REVIEW

**Review by keichwa — APPROVED** [review]
Thanks, lgtm!

When merged, I'll apply some minor layout and code formatting changes.

I'm wondering whether you can upgrade directly from suma 4.0 or even 3.2 to 4.2?  Maybe, we also must check the general server upgrade sections in this guide.  But lets keep the PR as small as possible ;)

File: modules/upgrade/pages/db-migration-13.adoc [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+====
+
+PostgreSQL upgrades can be performed in two ways: a regular upgrade, or a fast upgrade:
+
+A regular upgrade will create a complete copy of the database, so you will need double the existing database size of space available.
+Regular upgrades can take a considerable amount of time, depending on the size of the database and the speed of the storage system.
+
+A fast upgrade only takes a few minutes, and uses almost no additional disk space.
+However, if a fast upgrade fails, you must restore the database from the backup.
+A fast upgrade reduces the risk of running out of disk space.
```
Comment (mcalmer) [inline modules/upgrade/pages/db-migration-13.adoc:79]: ... but increase the risk of data lose when a backup does not exist or cannot be replayed

File: modules/upgrade/pages/db-migration-13.adoc [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    Without a verified database backup, you must not initiate a fast upgrade.
+    For more information on backing up, see xref:administration:backup-restore.adoc[].
+. Start the upgrade:
++
+----
+/usr/lib/susemanager/bin/pg-migrate-12-to-13.sh fast
+----
+. When the upgrade has successfully completed, you can safely delete the old database directory and reclaim lost disk space.
+    The old directory is renamed to [path]``/var/lib/pgsql/data-pg12``.
+// CHECKIT! ====>>>====>>>=======================================> ^^^^
```
Comment (mcalmer) [inline modules/upgrade/pages/db-migration-13.adoc:123]: @jordimassaguerpla will also the "fast" upgrade create such a directory? I thought it is more an implace upgrade. Would you check this please?

File: modules/upgrade/pages/server-major-upgrade-uyuni.adoc [hunk 3]
```diff
@@ -36,6 +36,25 @@ zypper up susemanager
...
+----
++
+For more information about PostgreSQL database migration, see xref:db-migration-13.adoc[].
+
+[NOTE]
+====
+Postgresql 12 or postgresql 13?
+
+This depends on the latest available version on the underlying operating system.
+Leap 15.2 has version 12 as the latest available version. 15.3 is expected to have postgresql 13.
```
Comment (mcalmer) [inline modules/upgrade/pages/server-major-upgrade-uyuni.adoc:52]: Hmm, my expectation is: with Leap 15.2 we will stay on postgresql 12 even if version 13 gets released there.
At the point in time when Uyuni switch to Leap 15.3 this DB migration will be required. But that will happen in about 6 month.
Not sure how to lower the user expectations here. We might need to check also with @paususe if we want to introduce pg13 in Uyuni earlier to get test results.

File: modules/upgrade/pages/db-migration-13.adoc (hunk 2 above)
Comment (jordimassaguerpla) [reply modules/upgrade/pages/db-migration-13.adoc:124]: fixed

File: modules/upgrade/pages/server-major-upgrade-uyuni.adoc (hunk 3 above)
Comment (jordimassaguerpla) [reply modules/upgrade/pages/server-major-upgrade-uyuni.adoc]: I've rephrase it to state that we use pg12 with leap 15.2, instead of saying we use the latest version available.

File: modules/upgrade/pages/db-migration-13.adoc (hunk 2 above)
Comment (jordimassaguerpla) [reply modules/upgrade/pages/db-migration-13.adoc:124]: Yes it will create the directory. The "fast upgrade" only adds the "--links" option to the pg_update command, but the directory is created anyway by the migration script.

File: modules/upgrade/pages/db-migration-13.adoc [hunk 4]
```diff
@@ -0,0 +1,124 @@
+[[db-migration-13]]
+= Database Migration from Version 12 to 13
+
+This section covers upgrading the PostgreSQL database from version{nbsp}12 to version{nbsp}13.
+If you are already using PostgreSQL 13, you do not need to perform this migration.
+If you are using version 10, see xref:upgrade:db-migration-12.adoc[].
+If you are using an older version, such as version 9.6, see xref:upgrade:db-migration-10.adoc[].
+
+If you want to upgrade to the latest {productname} version, you must be using PostgreSQL version 12 or 13, depending on the underlying operating system. When running on SLES 15 SP3, version 13 should be used. When running on Leap 15.2, version 12 should be used.
```
Comment (Loquacity) [inline modules/upgrade/pages/db-migration-13.adoc:9]:
```suggestion
If you want to upgrade to the latest {productname} version, you must be using PostgreSQL version 12 or 13, depending on the underlying operating system.

If you are running SLES 15 SP3, use PostgreSQL 13. 
If you are running Leap 15.2, use PostgreSQL 12.
```

File: modules/upgrade/pages/db-migration-13.adoc [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+* {productname} 3.1.4, SLES 12 SP3, Postgres 9.6
+* {productname} 3.1.11, SLES 12 SP3, Postgres 9.6
+* {productname} 3.2.10, SLES 12 SP3, Postgres 9.6
+* {productname} 3.2.10, SLES 12 SP3, Postgres 10
+* {productname} 3.2.10, SLES 12 SP4, Postgres 10
+* {productname} 4.0.0, SLES 15 SP1, Postgres 10
+* {productname} 4.1.0, SLES 15 SP2, Postgres 12
+* {productname} 4.2.0, SLES 15 SP3, Postgres 13
+
```
Comment (Loquacity) [inline modules/upgrade/pages/db-migration-13.adoc:25]: I feel like this is actually useful information for the reader. Can we turn it into a table perhaps?

File: modules/upgrade/pages/db-migration-13.adoc [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+rpm -q smdba
+----
++
+// CHECKIT!
++
+PostgreSQL{nbsp}13 requires ``smdba`` version 1.7.6 or later.
+
+. Perform a database backup.
+    For more information on backing up, see xref:administration:backup-restore.adoc[].
+
```
Comment (Loquacity) [inline modules/upgrade/pages/db-migration-13.adoc:61]: Needs some whitespace adjustments, but should render OK.

File: modules/upgrade/pages/db-migration-13.adoc [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+== Upgrade PostgreSQL
+
+[WARNING]
+====
+Always create a database backup before performing a migration.
+====
+
+PostgreSQL upgrades can be performed in two ways: a regular upgrade, or a fast upgrade:
+
+A regular upgrade will create a complete copy of the database, so you will need double the existing database size of space available.
```
Comment (Loquacity) [inline modules/upgrade/pages/db-migration-13.adoc:74]:
```suggestion
A regular upgrade creates a complete copy of the database, so you need double the existing database size of space available.
```

File: modules/upgrade/pages/server-major-upgrade-uyuni.adoc [hunk 8]
```diff
@@ -36,6 +36,26 @@ zypper up susemanager
 +
 For more information about PostgreSQL database migration, see xref:db-migration-12.adoc[].
 
+. To migrate the database to PostgreSQL 13 run the migrate script:
```
Comment (Loquacity) [inline modules/upgrade/pages/server-major-upgrade-uyuni.adoc:39]:
```suggestion
. To migrate the database to PostgreSQL 13 run the migration script:
```

**Review by Loquacity — CHANGES_REQUESTED** [review]

**Review by jcayouette — APPROVED** [review]
Approved once the edits requested by @Loquacity are included.

File: modules/upgrade/pages/db-migration-13.adoc (hunk 6 above)
Comment (jordimassaguerpla) [reply modules/upgrade/pages/db-migration-13.adoc:64]: which whitespaces do you refer to?

File: modules/upgrade/pages/db-migration-13.adoc (hunk 5 above)
Comment (jordimassaguerpla) [reply modules/upgrade/pages/db-migration-13.adoc:28]: I am not that familiar with asciidoc and actually I don't know how to render it to test it.

How would a table look like in asciidoc?

Comment (jordimassaguerpla) [reply modules/upgrade/pages/db-migration-13.adoc:28]: Are we talking about this?

```
[options="header"]
|=================================================
| SUSE Manager version | SLES version | Postgresql version
| {productname} 3.1.4 | SLES 12 SP3 | Postgres 9.6                                
| {productname} 3.1.11 | SLES 12 SP3 | Postgres 9.6                               
| {productname} 3.2.10 | SLES 12 SP3 | Postgres 9.6                               
| {productname} 3.2.10 | SLES 12 SP3 | Postgres 10                                
| {productname} 3.2.10 | SLES 12 SP4 | Postgres 10                                
| {productname} 4.0.0 | SLES 15 SP1 | Postgres 10                                 
| {productname} 4.1.0 | SLES 15 SP2 | Postgres 12                                 
| {productname} 4.2.0 | SLES 15 SP3 | Postgres 13
|=================================================
```

Comment (Loquacity) [reply modules/upgrade/pages/db-migration-13.adoc:28]: You can look at the install guide requirements chapter, the client cfg guide, or the reference guide to see examples, but this is the basic syntax:

```
[[ref-id]]
.Title
[cols="1,1,1", options="header"]
|===
| Column 1                  | Column 2                 | Column 3
| Content in column 1 | Content in column 2 | Content in column 3
|=== 
```

All good if you don't want to do this, though, just raise a new issue and I'll sort it out :)

File: modules/upgrade/pages/db-migration-13.adoc (hunk 6 above)
Comment (Loquacity) [reply modules/upgrade/pages/db-migration-13.adoc:64]: No need to put newlines between steps, and we usually put three newlines before a title. But it's just cosmetic, so no big deal.

File: modules/upgrade/pages/db-migration-13.adoc (hunk 5 above)
Comment (jordimassaguerpla) [reply modules/upgrade/pages/db-migration-13.adoc:28]: Should the ref-id be "ref-id" or be some kind of id? If so, which?

I have this now

`
////                                                                            
== Background Information                                                       
[[ref-id]]                                                                      
[cols="1,1,1", options="header"]                                                
|=================================================                              
| {productname} version | SLES version | Postgresql version                     
| {productname} 3.1.4 | SLES 12 SP3 | Postgres 9.6                                
| {productname} 3.1.11 | SLES 12 SP3 | Postgres 9.6                               
| {productname} 3.2.10 | SLES 12 SP3 | Postgres 9.6                               
| {productname} 3.2.10 | SLES 12 SP3 | Postgres 10                                
| {productname} 3.2.10 | SLES 12 SP4 | Postgres 10                                
| {productname} 4.0.0 | SLES 15 SP1 | Postgres 10                                 
| {productname} 4.1.0 | SLES 15 SP2 | Postgres 12                                 
| {productname} 4.2.0 | SLES 15 SP3 | Postgres 13                               
|=================================================                              
                                                                                
                                                                                
////
`

File: modules/upgrade/pages/db-migration-13.adoc (hunk 6 above)
Comment (jordimassaguerpla) [reply modules/upgrade/pages/db-migration-13.adoc:64]: jejeje, what is "steps" and what is "title" ?

Comment (Loquacity) [reply modules/upgrade/pages/db-migration-13.adoc:64]: Let's just merge this as-is, and I'll fix the table in a new PR.

**Review by Loquacity — APPROVED** [review]
